pkg/module/drift/terraform.go

```diff
@@ -2,19 +2,85 @@ package drift
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"os"
 	"reflect"
 	"strings"
+	"time"
 
+	"github.com/cloudquery/cloudquery/pkg/module/drift/terraform"
 	"github.com/cloudquery/cq-provider-sdk/provider/schema"
 	"github.com/doug-martin/goqu/v9"
 	"github.com/doug-martin/goqu/v9/exp"
 	"github.com/jackc/pgx/v4/pgxpool"
 	"github.com/olekukonko/tablewriter"
 )
 
-func (d *Drift) driftTerraform(ctx context.Context, conn *pgxpool.Conn, cloudName string, cloudTable *traversedTable, resName string, resources map[string]*ResourceConfig, iacData *IACConfig) (*Result, error) {
+type TFStates []*terraform.Data
+
+// FindType returns all instances of the given type under a given mode
+func (t TFStates) FindType(tfType, tfMode string) TFInstances {
```

Comment:
`tfMode` filter might be removed because some resources are kept in `data` mode anyway... in the future we might even have a "this resource is stored under data" flag/whatever in drift.hcl terraform block maybe? I was going to conclude this comment with "it's not efficient use of time", but since it's going to stay one way or another, I'll put it in.